sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DecimalPrecision.scala

```diff
@@ -60,7 +56,7 @@ import org.apache.spark.sql.types._
  */
 // scalastyle:on
 object DecimalPrecision extends TypeCoercionRule {
```

Comment:
We should update the calssdoc of this rule. Now it only cast non-decimal to decimal type.

Reply:
not all, I leave BinaryComparison
```scala
    case b @ BinaryComparison(e1 @ DecimalType.Expression(p1, s1),
    e2 @ DecimalType.Expression(p2, s2)) if p1 != p2 || s1 != s2 =>
      val resultType = widerDecimalType(p1, s1, p2, s2)
      val newE1 = if (e1.dataType == resultType) e1 else Cast(e1, resultType)
      val newE2 = if (e2.dataType == resultType) e2 else Cast(e2, resultType)
      b.makeCopy(Array(newE1, newE2))
```

Reply:
OK but we still need to update the classdoc, right?

Reply:
at least binary arithmetics are not handled by this rule now.

Reply:
add a docs for the binary arithmetics